ip.go

```diff
@@ -5,11 +5,27 @@ import (
 	"strings"
 )
 
-// Ranges of addresses allocated by IANA for private internets, as per RFC1918.
+// Ranges of addresses allocated by IANA for private internets, as per RFC1918 and RFC3927
+// for Link-local addresses.
 var PrivateNetworks = []string{
 	"10.0.0.0/8",
 	"172.16.0.0/12",
+	"169.254.0.0/16",
 	"192.168.0.0/16",
+	"fe80::/10",
+	"fd00::/8",
+}
+
+var privateNets []*net.IPNet
```

Comment:
`net.ParseCIDR`: https://golang.org/pkg/net/#ParseCIDR

`net.ParseCIDR` returns a pointer `*net.IPNet`, so thats what I made the slice of.  It looks like we could make it a non-pointer, but the practical effects seem near nil.